libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,623 @@
+        url: str,
+        username: str,
+        password: str,
+        metadatas: Optional[List[dict]] = None,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        ids: Optional[List[str]] = None,
+        pre_delete_collection: bool = False,
+        **kwargs: Any,
+    ) -> Neo4jVector:
+        """Construct Neo4jVector wrapper from raw documents and pre-
+        generated embeddings.
+
+        Return Neo4jVector initialized from documents and embeddings.
+        Neo4j credentials are required in the form of
+        `url`, `username`, and `password` parameters.
+
+        Example:
+            .. code-block:: python
+
+                from langchain.vectorstores.neo4j_vector import Neo4jVector
+                from langchain.embeddings import OpenAIEmbeddings
+                embeddings = OpenAIEmbeddings()
+                text_embeddings = embeddings.embed_documents(texts)
+                text_embedding_pairs = list(zip(texts, text_embeddings))
+                vectorstore = Neo4jVector.from_embeddings(
+                    text_embedding_pairs, embeddings)
+        """
+        texts = [t[0] for t in text_embeddings]
+        embeddings = [t[1] for t in text_embeddings]
+
+        return cls.__from(
+            texts,
+            embeddings,
+            embedding,
+            url=url,
+            username=username,
+            password=password,
+            metadatas=metadatas,
+            ids=ids,
+            distance_strategy=distance_strategy,
+            pre_delete_collection=pre_delete_collection,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_existing_index(
+        cls: Type[Neo4jVector],
+        embedding: Embeddings,
+        index_name: str,
+        username: str,
+        password: str,
+        url: str,
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        **kwargs: Any,
+    ) -> Neo4jVector:
+        """
+        Get instance of an existing Neo4j vector index.This method will
+        return the instance of the store without inserting any new
+        embeddings.
+        Neo4j credentials are required in the form of
+        `url`, `username`, and `password` parameters along with
+        the `index_name` definition.
+        """
+
+        store = cls(
+            username=username,
+            password=password,
+            url=url,
+            embedding_function=embedding,
+            index_name=index_name,
+            distance_strategy=distance_strategy,
+            **kwargs,
+        )
+
+        embedding_dimension = store.retrieve_existing_index()
+
+        if not embedding_dimension:
+            raise ValueError(
+                "The specified vector index name does not exist. "
+                "Make sure to check if you spelled it correctly"
+            )
+
+        # Check if embedding function and vector index dimensions match
+        if not store.embedding_dimension == embedding_dimension:
```

Comment:
also check for embedding function

Reply:
that's what we are doing. As mentioned, the only way to get embedding dimension in LangChain is to have the embedding function embed a text: 

https://github.com/langchain-ai/langchain/pull/9770/files/87b2c5a165549dff5c22c2280f561af677768d7a#diff-c6a7dc81163d2af03aa6e6d40a2094336f612ed2ba7dfd80248b8c6cfc6423ebR122